tlsutil/tlsgen.go

```diff
@@ -0,0 +1,129 @@
+package tlsutil
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func CreateLocalTLSData(identityName string) error {
+
+	if len(identityName) == 0 {
+		identityName = "test"
+	}
+
+	caKey := exec.Command("openssl", "genrsa", "-out", "ca.key", "2048")
+	err := caKey.Run()
+	if err != nil {
+		return err
+	}
+	caCert := exec.Command("openssl", "req", "-x509", "-new", "-nodes", "-key", "ca.key", "-sha256", "-days", "1", "-out", "ca.crt", `-subj`, `/C=US/ST=California/L=Alameda/O=XLB`)
+	err = caCert.Start()
+	if err != nil {
+		return fmt.Errorf("cannot generate caCert, error: %w", err)
+	}
+	err = caCert.Wait()
+	if err != nil {
+		return err
+	}
+
+	// Server TLS
+	srvKey := exec.Command("openssl", "genrsa", "-out", "server.key", "2048")
+	err = srvKey.Start()
+	if err != nil {
+		return err
+	}
+	err = srvKey.Wait()
+	if err != nil {
+		return err
+	}
+
+	srvCSR := exec.Command("openssl", "req", "-new", "-key", "server.key", "-out", "server.csr", "-subj", "/C=US/ST=California/L=Alameda/O=XLB")
+	err = srvCSR.Start()
+	if err != nil {
+		return err
+	}
+	err = srvCSR.Wait()
+	if err != nil {
+		return err
+	}
+
+	if err := os.WriteFile("server_san.txt", []byte("subjectAltName=DNS:localhost"), 0666); err != nil {
+		return err
+	}
+
+	srvCert := exec.Command("openssl", "x509", "-req", "-in", "server.csr", "-CA", "ca.crt", "-CAkey", "ca.key", "-CAcreateserial", "-out", "server.crt", "-days", "1", "-sha256", "-extfile", "server_san.txt")
+	err = srvCert.Start()
```

Comment:
Nit: Calls of Start followed by Wait could likely be replaced by a single call to Run

Reply:
I was unsuccessful with `exec.Run` on my machine for those commands.